python/benchmark/test_gen_data.py

```diff
@@ -238,15 +286,38 @@ def test_make_sparse_regression(
             # assert sparseVec.toArray().dtype == np.dtype(dtype), "Unexpected dtype"
             assert sparseVec.size == col_num, "X col number mismatch"
         assert y.shape == (row_num,), "y shape mismatch"
-        assert c.shape == (col_num,), "coef shape mismatch"
-        assert sum(c != 0.0) == 3, "Unexpected number of informative features"
+
+        if logistic_regression == "False" or n_classes_num == 2:
+            assert c.shape == (col_num,), "coef shape mismatch"
+            assert np.count_nonzero(c) == 3, "Unexpected number of informative features"
+        else:
+            print(
+                c.shape,
+                (
+                    n_classes,
+                    col_num,
+                ),
+            )
+            assert c.shape == (
+                n_classes_num,
+                col_num,
+            ), "coef shape mismatch"
+            assert (
+                np.count_nonzero(c) == 3 * n_classes_num
+            ), "Unexpected number of informative features"
 
         X_np = np.array([r.toArray() for r in X])
 
         if logistic_regression == "True":
-            # Test that X consists of only 0 or 1
+            # Test that X consists of only discrete label
+            possible_labels = range(n_classes_num)
             for n in y:
-                assert n == 0 or n == 1
+                found = False
+                for l in possible_labels:
+                    if n == l:
+                        found = True
+                        break
+                assert found, "Invalid label"
```

Comment:
Can we check unique(y) == n_classes

Reply:
There is the possibility that during randomness we lose completely a class even if less likely given sufficient data samples.